Sources_v3/Controllers/ChannelController/ChannelController.swift

```diff
@@ -575,6 +575,17 @@ public extension ChannelControllerGeneric {
             }
         }
     }
+    
+    /// Marks the channel as read.
+    /// - Parameter completion: The completion. Will be called on a **callbackQueue** when the network request is finished.
+    ///                         If request fails, the completion will be called with an error.
+    func markRead(completion: ((Error?) -> Void)? = nil) {
+        worker.markRead(cid: channelId) { [weak self] error in
+            self?.callback {
+                completion?(error)
+            }
+        }
+    }
```

Comment:
This should be tested. You also need to guard for `isChannelAlreadyCreated` because it's not possible to mark read channel that doesn't exist yet.

Reply:
We should leave the guarding part out IMHO. Backend error response will be plenty enough to debug, and we'd like to prevent failures on `client.channelController(cid: cid).markRead()`

Reply:
Fixed in d88845270f2dfe90ce3f887da7d256f728e1fb0e

Reply:
tip: You need to commit first and then mention the hash, otherwise GitHub won't make a link out of the hash.